keps/sig-auth/2579-psp-replacement/README.md

```diff
@@ -0,0 +1,1083 @@
+3. Grant (real) users that should have privileged debug capability the ability to impersonate the
+   exempt user.
+
+We could consider ways to streamline the user experience of this, for instance adding a special RBAC
+binding that exempts users when operating on the ephemeral containers subresource (e.g. an
+`escalate-privilege` verb on the ephemeral containers subresource).
+
+<<[/UNRESOLVED]>>
+
+#### Other Pod Subresources
+
+Aside from ephemeral containers, the policy is not checked for any other Pod subresources (status,
+bind, logs, exec, attach, port-forward).
+
+Although annotations can be updated through the status subresource, the apparmor annotations are
+immutable and the seccomp annotations are deprecated and slated for removal in v1.23.
+
+### Pod Security Standards
+
+Policy level definitions are hardcoded and unconfigurable out of the box. However, the [Pod Security
+Standards] leave open ended guidance on a few items, so we must make a static decision on how to
+handle these elements:
+
+**HostPorts** - (baseline) HostPorts will be forbidden. This is a more niche feature, and violates
+the container-host boundary.
+
+**AppArmor** - (baseline) Allow anything except `unconfined`. Custom AppArmor profiles must be
+installed by the cluster admin, and AppArmor fails closed when a profile is not present.
+
+**SELinux** - (baseline) type may only be set to allowlisted values, level may be anything, user &
+role must be unset. Spec:
+
+- SELinuxOptions.Type
+    - **Restricted Fields:**
+        - `spec.securityContext.seLinuxOptions.type`
+        - `spec.containers[*].securityContext.seLinuxOptions.type`
+        - `spec.initContainers[*].securityContext.seLinuxOptions.type`
+    - **Allowed Values:**
+        - undefined/empty
+        - `container_t`
+        - `container_init_t`
+        - `container_kvm_t`
+
+- SELinuxOptions.User and SELinuxOptions.Role
+    - **Restricted Fields:**
+        - `spec.securityContext.seLinuxOptions.user`
+        - `spec.containers[*].securityContext.seLinuxOptions.user`
+        - `spec.initContainers[*].securityContext.seLinuxOptions.user`
+        - `spec.securityContext.seLinuxOptions.role`
+        - `spec.containers[*].securityContext.seLinuxOptions.role`
+        - `spec.initContainers[*].securityContext.seLinuxOptions.role`
+    - **Allowed Values:** undefined/empty
+
+- SELinuxOptions.Level
+    - Unrestricted.
+
+**Non-root Groups** - (restricted) This optional constraint will be omitted from the initial
+implementation.
+
+**Seccomp**
+- (restricted) Must be set to anything except `unconfined`. Same reasoning as AppArmor. If the default
+  value changes from `unconfined` (see https://github.com/kubernetes/enhancements/issues/2413), then
+  the requirement to set a profile will be lifted.
+- (baseline) May be unset; any profile except `unconfined` allowed.
+
+**Capabilities** - (baseline) Only the following capabilities may be added:
```

Comment:
restricted could still use the capabilities when non-root by using file capabilities.

@liggitt @tallclair

Reply:
no_new_privs does prevent that though.

Reply:
I don't think so, since we also require `AllowPrivilegeEscalation=false` (aka `no_new_privs`) for restricted.